tests/python/pipeline/sieves/test_os_sieve.py

```diff
@@ -0,0 +1,215 @@
+                version="==1.9.0",
+                index=Source("https://tensorflow.pypi.thoth-station.ninja/index/rhel/7.6/jemalloc/simple/"),
+                develop=False,
+            ),
+            PackageVersion(
+                name="pytest",
+                version="==3.0.0",
+                index=pypi_source,
+                develop=True,
+            ),
+        ]
+
+    def test_noop_no_os(self):
+        """Test no changes are made if no specific operating system is configured."""
+        package_versions = self._get_packages_no_aicoe()
+        sieve_context = SieveContext.from_package_versions(package_versions)
+
+        # Do not assign runtime environment intentionally - it will default to no environment.
+        project = Project.from_strings(pipfile_str=self._PIPFILE_CONTENT_NO_AICOE)
+        os_sieve = OperatingSystemSieve(graph=None, project=project)
+        os_sieve.run(sieve_context)
+
+        assert {pv.to_tuple() for pv in package_versions} == set(sieve_context.iter_direct_dependencies_tuple())
+
+    def test_noop_no_aiceo(self):
+        """Test no changes are made if no AICoE releases are found."""
+        package_versions = self._get_packages_no_aicoe()
+        sieve_context = SieveContext.from_package_versions(package_versions)
+
+        # Do not assign runtime environment intentionally - it will default to no environment.
+        project = Project.from_strings(pipfile_str=self._PIPFILE_CONTENT_NO_AICOE)
+        os_sieve = OperatingSystemSieve(graph=None, project=project)
+        os_sieve.run(sieve_context)
+
+        assert {pv.to_tuple() for pv in package_versions} == set(sieve_context.iter_direct_dependencies_tuple())
+
+    def test_os_sieve(self):
+        """Test removal of packages based on AICoE package source index configuration.
+
+        We keep only TensorFlow release which is from PyPI and manylinux2010 build as there is no match on OS release.
+        """
+        package_versions = self._get_packages_aicoe()
+        sieve_context = SieveContext.from_package_versions(package_versions)
+
+        # Do not assign runtime environment intentionally - it will default to no environment.
+        project = Project.from_strings(
+            pipfile_str=self._PIPFILE_CONTENT_AICOE,
+            runtime_environment=RuntimeEnvironment.from_dict({
+                "operating_system": {
+                    "name": "rhel",
+                    "version": "7.5"
+                }
+            })
+        )
+        os_sieve = OperatingSystemSieve(graph=None, project=project)
+        os_sieve.run(sieve_context)
+
+        expected = {
+            ("pytest", "3.0.0", "https://pypi.org/simple"),
+            ("tensorflow", "1.9.0", "https://pypi.org/simple"),
+            # Filtering out this entry is left on another sieve which ensures runtime environment compatibility.
+            ("tensorflow", "1.9.0", "https://tensorflow.pypi.thoth-station.ninja/index/manylinux2010/jemalloc/simple/"),
```

Comment:
we don't have tensorflow==1.9.0 for manylinux2010 in our index yet.

Reply:
I just put some value - these values are artificial for test cases, not prod code.

Reply:
ah, thanks for the clarification.